Unnecessary ',' is contained in the npm command.

## Project

**Name**: Virtual Assistant <!--Virtual Assistant, Skills, Enterprise Bot Template, or Customer Support Template-->

**Language**: TypeScript <!-- C#, TypeScript -->

## Description
<!--A clear and concise description of what the bug is.-->

[Quickstart: Create your first Virtual Assistant (TypeScript)](https://github.com/Microsoft/AI/blob/master/docs/tutorials/typescript/virtualassistant.md#download-and-install) - Download and install 

Running step 3 will result in an error.

## To Reproduce
<!--Steps to reproduce the behavior:-->

```cmd
C:\>npm install -g botdispatch, ludown, luis-apis, qnamaker, luisgen
npm ERR! code EINVALIDTAGNAME
npm ERR! Invalid tag name "botdispatch,": Tags may not have any characters that encodeURIComponent encodes.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\hiroyha.FAREAST\AppData\Roaming\npm-cache\_logs\2019-05-04T16_13_42_709Z-debug.log
```

## Expected behavior
<!--A clear and concise description of what you expected to happen.-->

The correct command is the following:

```cmd
npm install -g botdispatch ludown luis-apis qnamaker luisgen
```

--- a/docs/howto/skills/addingskills.md
+++ b/docs/howto/skills/addingskills.md
@@ -1,19 +1,28 @@
+# Adding Skills
+
 ## Pre-requisites
 - [Node.js](https://nodejs.org/) version 10.8 or higher
-- Install the [Azure Command Line Tools (CLI)](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?view=azure-cli-latest)
+- Dispatch CLI which is used as part of skill registration
+    ```
+    npm install -g botdispatch
+    ```
 
+## Adding Skills
 
-## Installation
-To install using npm
+We have two approaches to add Skills to your Virtual Assistant which can be used interchangeably.
+
+1. The first leverages a `add_remote_skill.ps1` Powershell script included as part of your Virtual Assistant solution (located in the `Deployment\Scripts` folder of your assistant).
+2.  `botskills` command line tool which is in preview.
+
+> Skills requiring authentication (e.g. the productivity skills) should be added using the Powershell script at this time.
+
+If you wish to use the `botskills` CLI then you can install using the following npm command:
+
 ```bash
 npm install -g botdispatch, botskills
 ```
-This will install botdispatch and botskills into your global path.
-To uninstall using npm
-```bash
-npm uninstall -g botdispatch, botskills
-```
-> Your Virtual Assistant must have been deployed using the [deployment tutorial](/docs/tutorials/assistantandskilldeploymentsteps.md) before using the `botskills` tool as it relies on the Dispatch models being available and a deployed Bot for authentication connection information.
+
+> Your Virtual Assistant must have been deployed using the [deployment tutorial](/docs/tutorials/assistantandskilldeploymentsteps.md) before using the add_remote_skill `botskills` tool as it relies on the Dispatch models being available and a deployed Bot for authentication connection information.
 
 ## Skill Deployment
 
@@ -21,7 +30,7 @@ See the [Skills Overview](/docs/README.md#skills) section for details on the Ski
 
 ## Skill CLI 
 
-The Skill CLI provides automation of all key steps required to add a Skill to your project
+The Powershell script and CLI provides automation of all key steps required to add a Skill to your project
 
 1. Retrieve the Skill Manifest from the remote Skill through the `/api/skill/manifest` endpoint.
 2. Identify which Language Models are used by the Skill and resolve the triggering utterances either through local LU file resolution or through inline trigger utterances if requested.
@@ -33,8 +42,14 @@ The Skill CLI provides automation of all key steps required to add a Skill to yo
 
 Run the following command to add each Skill to your Virtual Assistant. This assumes you are running the CLI within the project directory and have created your Bot through the template and therefore have a `skills.json` file present.
 
-The `--luisFolder` parameter can be used to point the Skill CLI at the source LU files for trigger utterances. For Skills provided within this repo these can be found in the `Deployment\Resources\LU` folder of each Skill. The CLI will automatically traverse locale folder hierarchies.
+The `--luisFolder` parameter can be used to point the Skill CLI at the source LU files for trigger utterances. For Skills provided within this repo these can be found in the `Deployment\Resources\LU` folder of each Skill. The CLI will automatically traverse locale folder hierarchies.  This can be omitted for any of the skills we provide as the LU files are provided locally.
 
+- Powershell:
+```
+.\Deployment\scripts\add_remote_skill.ps1 -botName "YOUR_BOT_NAME" -manifestUrl https://YOUR_SKILL.azurewebsites.net/api/skill/manifest -luisFolder [path]
+```
+
+- botskills CLI:
 ```bash
 botskills connect --botName YOUR_BOT_NAME --remoteManifest "http://<YOUR_SKILL_MANIFEST>.azurewebsites.net/api/skill/manifest" --luisFolder [path] --cs 
 ```
@@ -62,6 +77,12 @@ For Skills that require other Authentication connection configuration please fol
 
 To disconnect a skill from your Virtual Assistant use the following command, passingthe id of the Skill as per the manifest (e.g. calendarSkill).
 
+- Powershell:
+```
+.\Deployment\scripts\remove_skill.ps1 -manifestUrl https://YOUR_SKILL.azurewebsites.net/api/skill/manifest
+```
+
+botskills:
 ```bash
 botskills disconnect --skillId SKILL_ID
 ```

--- a/docs/readme.md
+++ b/docs/readme.md
@@ -1,12 +1,13 @@
 ![Bot Framework Solutions](/docs/media/bot_framework_solutions_header.png)
  
 # Table of Contents
+- [Table of Contents](#table-of-contents)
 - [Overview](#overview)
 - [Tutorials](#tutorials)
-- [How-To](#advanced-scenarios)
+- [How-To](#how-to)
   - [Virtual Assistant](#virtual-assistant)
   - [Skills](#skills)
-- [Reference](#reference-documentation)
+- [Reference](#reference)
   - [Virtual Assistant](#virtual-assistant-1)
   - [Skills](#skills-1)
   - [Analytics](#analytics)
@@ -31,7 +32,7 @@
 |1. Quickstart: Create your first Virtual Assistant<br/><br/> <p align="center">[![Quickstart: Create a Virtual Assistant with C#](./media/csharp_icon.png)](/docs/tutorials/csharp/virtualassistant.md)[![Quickstart: Create a Virtual Assistant with TypeScript](./media/typescript_icon.png)](/docs/tutorials/typescript/virtualassistant.md)</p>|Deploying your own assistant using the Virtual Assistant template|
 |2. Customize your Virtual Assistant <br/><br/><p align="center">[![Customize your Virtual Assistant with C#](./media/csharp_icon.png)](/docs/tutorials/csharp/customizeassistant.md) [![Customize your Assistant with TypeScript](./media/typescript_icon.png)](/docs/tutorials/typescript/customizeassistant.md)</p>|Personalize your assistant, change the name, branding, QnA| 
 |3. Create a new Skill <br/><br/><p align="center">[![Create and deploy a new Skill with C#](./media/csharp_icon.png)](/docs/tutorials/csharp/skill.md) [![Create and deploy a new Skill with TypeScript](./media/typescript_icon.png)](/docs/tutorials/typescript/skill.md)</p>|Creating a new Skill using the template|
-|4. Customize your Skill <br/><br/><p align="center">[![Customize your Skill with C#](./media/csharp_icon.png)](/docs/tutorials/csharp/customizeskill.md) [![Customize your Skill with TypeScript](./media/typescript_icon.png)](/docs/tutorials/typescript/customizeassistant.md)</p>|Building your first skill| 
+|4. Customize your Skill <br/><br/><p align="center">[![Customize your Skill with C#](./media/csharp_icon.png)](/docs/tutorials/csharp/customizeskill.md)</p>|Building your first skill| 
 
 # How-To
 
@@ -41,7 +42,6 @@
 
 | Name | Description |
 | ---- | ----------- |
-| How To: Authenticate users in your Assistant <br/><br/><p align="center">[![Authenticating users to your Assistant with C#](./media/csharp_icon.png)](/docs/howto/assistant/csharp/addauthentication.md) [![Authenticating users to your Assistant with TypeScript](./media/typescript_icon.png)](/docs/howto/assistant/typescript/addauthentication.md)</p>|How to add user authentication to your Assistant |
 | [How To: Enable Linked Accounts](/docs/howto/assistant/linkedaccounts.md) | Enable users to link third party accounts (e.g. Office 365) to your Assistant |
 | How To: Migrate from the Enterprise Template <br/><br/><p align="center">[![Migrating from the Enterprise Template with C#](./media/csharp_icon.png)](/docs/howto/assistant/csharp/ettovamigration.md)</p> | Guidance on how to move from an Enterprise Template based Bot to the new Template|
 | How to: Migrate from the old Virtual Assistant solution <br/><br/><p align="center">[![Migrating from the beta Virtual Assistant solution with C#](./media/csharp_icon.png)](/docs/howto/assistant/csharp/oldvatovamigration.md)</p>|Guidance on how to move from the original Virtual Assistant solution to the new Template |
@@ -49,7 +49,6 @@
 | [How To: Enable cross bot communication into one conversational experience](/docs/howto/assistant/parentchildbotpattern.md) | Create one central Assistant which hands-off to child bots (a common enterprise scenario) |
 | [How To: Customize Azure Resource Deployment](/docs/howto/assistant/customizedeployment.md) | How to customise the provided ARM template for different deployment scenarios. |
 | How To: Secure your keys using Azure Key Vault :construction_worker_woman: | How to safeguard your keys using Azure Key Vault|
-
 
 ## Skills
 
@@ -68,7 +67,7 @@
 
 | Name | Description |
 | ---- | ----------- |
-|[Template Outline](/docs/reference/assistant/templateoutline.md)|An outline of what the Virtual Assistant template provides| 
+|[Template Outline](/docs/reference/assistant/templateoutline.md)|An outline of what the Virtual Assistant template provides|
 |[Under the covers](/docs/reference/assistant/underthecovers.md)|Detailed documentation covering what the template provides and how it works| 
 |[Underlying architecture](/docs/reference/assistant/architecture.md)|Detailed exploration of the overall Virtual Assistant Architecture|
 |[Template structure](/docs/reference/assistant/projectstructure.md)|Walkthrough of your Virtual Assistant project|

--- a/docs/reference/skills/experimental.md
+++ b/docs/reference/skills/experimental.md
@@ -38,6 +38,6 @@ The Experimental Skills require the following dependencies for end to end operat
 
 > Review the pricing and terms for the services and adjust to suit your scenario.
 
-To deploy the experimental skills using the default configuration, follow the steps in this common [deployment documentation page](/docs/virtual-assistant/common/deploymentsteps.md) from the folder where your have cloned the GitHub repo.
+To deploy the experimental skills using the default configuration, follow the steps in this common [deployment documentation page](/docs/tutorials/assistantandskilldeploymentsteps.md) from the folder where your have cloned the GitHub repo.
 
 

--- a/docs/reference/skills/productivity-calendar.md
+++ b/docs/reference/skills/productivity-calendar.md
@@ -130,7 +130,7 @@ The Calendar Skill require the following dependencies for end to end operation w
 
 > Review the pricing and terms for the services and adjust to suit your scenario.
 
-To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/virtual-assistant/common/deploymentsteps.md) from the folder where your have cloned the GitHub repo.
+To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/tutorials/assistantandskilldeploymentsteps.md) from the folder where your have cloned the GitHub repo.
 
 ### Authentication Connection Settings
 

--- a/docs/reference/skills/productivity-email.md
+++ b/docs/reference/skills/productivity-email.md
@@ -109,7 +109,7 @@ The Email Skill require the following dependencies for end to end operation whic
 
 > Review the pricing and terms for the services and adjust to suit your scenario.
 
-To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/virtual-assistant/common/deploymentsteps.md) from the folder where your have cloned the GitHub repo.
+To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/tutorials/assistantandskilldeploymentsteps.md) from the folder where your have cloned the GitHub repo.
 
 ### Authentication Connection Settings
 

--- a/docs/reference/skills/productivity-todo.md
+++ b/docs/reference/skills/productivity-todo.md
@@ -78,7 +78,7 @@ The ToDo Skill require the following dependencies for end to end operation which
 
 > Review the pricing and terms for the services and adjust to suit your scenario.
 
-To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/virtual-assistant/common/deploymentsteps.md) from the folder where your have cloned the GitHub repo.
+To deploy your services using the default configuration, follow the steps in this common [deployment documentation page](/docs/tutorials/assistantandskilldeploymentsteps.md) from the folder where your have cloned the GitHub repo.
 
 ### Authentication Connection Settings
 Your Authentication Connection and corresponding Application Registration should have the following Scopes added, these will be added automatically as part of Skill configuration where possible.

--- a/docs/tutorials/assistantandskilldeploymentsteps.md
+++ b/docs/tutorials/assistantandskilldeploymentsteps.md
@@ -22,7 +22,16 @@
     parametersFile | A .json file that can overwrite the default values of the Azure Resource Manager template. | No
     outFolder | Output directory for created appsettings.json and cognitivemodels.json files. Default value is current directory. | No
 
-> There is a known issue with some users whereby you might experience the following error when running deployment `Could not provision Microsoft App Registration automatically. Please provide the -appId and -appPassword arguments for an existing app and try again`. In this situation, please create your own [Azure Active Directory App](https://ms.portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Overview) and manually create a new application retrieving the ApplicationID and Password/Secret. Run the above deployment script again but provide two new arguments `appId` and `appPassword` passing the values you've just retrieved.
+> There is a known issue with some users whereby you might experience the following error when running deployment `Could not provision Microsoft App Registration automatically. Please provide the -appId and -appPassword arguments for an existing app and try again`. In this situation, please create your own Azure Active Directory App through [this](https://ms.portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredAppsPreview) part of the Azure portal.
+- Click `New Registration`
+- Provide a name
+- Select `Accounts in any organizational directory`
+
+Then once created retrieve the `Application (ClientId)` and create a new client secret on the `Certificates & secrets` pane
+
+Run the above deployment script again but provide two new arguments `appId` and `appPassword` passing the values you've just retrieved.
+
+You also need to set
 
 > NOTE: Take special care when providing the appSecret step above as special characters (e.g. @) can cause parse issues. Ensure you wrap these parameters in single quotes.
 

--- a/docs/tutorials/assistantandskilldeploymentsteps.md
+++ b/docs/tutorials/assistantandskilldeploymentsteps.md
@@ -31,8 +31,6 @@ Then once created retrieve the `Application (ClientId)` and create a new client 
 
 Run the above deployment script again but provide two new arguments `appId` and `appPassword` passing the values you've just retrieved.
 
-You also need to set
-
 > NOTE: Take special care when providing the appSecret step above as special characters (e.g. @) can cause parse issues. Ensure you wrap these parameters in single quotes.
 
 ## Customising deployment using the parameters file

--- a/docs/tutorials/csharp/skill.md
+++ b/docs/tutorials/csharp/skill.md
@@ -3,16 +3,22 @@
 **APPLIES TO:** ✅ SDK v4
 
 ## In this tutorial
-- [Intro](#intro)
-- [Download and install](#download-and-install)
-- [Create your Skill](#create-your-skill)
-- [Deploy your Skill](#deploy-your-skill)
-- [Test your Skill](#test-your-skill)
-- [Update your Skill manifest](#update-your-skill-manifest)
-- [Publish your Skill](#publish-your-skill)
-- [Validate the Skill manifest endpoint](#validate-the-skill-manifest-endpoint)
-- [Adding your Skill to an assistant](#adding-your-skill-to-an-assistant)
-- [Testing your Skill](#testing-your-skill)
+- [Create a new Bot Framework Skill (C#)](#create-a-new-bot-framework-skill-c)
+  - [In this tutorial](#in-this-tutorial)
+  - [Intro](#intro)
+    - [Purpose](#purpose)
+    - [Prerequisites](#prerequisites)
+    - [Time to Complete](#time-to-complete)
+    - [Scenario](#scenario)
+  - [Download and install](#download-and-install)
+  - [Create your Skill](#create-your-skill)
+  - [Deploy your Skill](#deploy-your-skill)
+  - [Test your Skill](#test-your-skill)
+  - [Update your Skill manifest](#update-your-skill-manifest)
+  - [Publish your Skill](#publish-your-skill)
+  - [Validate the Skill manifest endpoint](#validate-the-skill-manifest-endpoint)
+  - [Adding your Skill to an assistant](#adding-your-skill-to-an-assistant)
+  - [Testing your Skill](#testing-your-skill)
 
 ## Intro
 ### Purpose
@@ -47,7 +53,7 @@ A Bot Framework Skill app (in C#) that greets a new user.
    * [Download PowerShell Core on macOS and Linux](https://aka.ms/getps6-linux)
 5. Install  Bot Framework (CLI) tool dependencies. It's important to do this even if you have earlier versions as we make use of the latest capabilities: 
    ```
-   npm install -g botdispatch, ludown, luis-apis, qnamaker, luisgen, botskills
+   npm install -g botdispatch ludown luis-apis luisgen botskills
    ```
 6. Install the [Azure Command Line Tools (CLI)](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?view=azure-cli-latest)
 

--- a/docs/tutorials/csharp/virtualassistant.md
+++ b/docs/tutorials/csharp/virtualassistant.md
@@ -3,7 +3,7 @@
 **APPLIES TO:** ✅ SDK v4
 
 ## In this tutorial
-- [Quickstart: Creating your first Virtual Assistant (C#)](#quickstart-creating-your-first-virtual-assistant-c)
+- [Quickstart: Create your first Virtual Assistant (C#)](#quickstart-create-your-first-virtual-assistant-c)
   - [In this tutorial](#in-this-tutorial)
   - [Intro](#intro)
     - [Purpose](#purpose)
@@ -50,7 +50,7 @@ A Virtual Assistant app (in C#) that greets a new user.
    * [Download PowerShell Core on macOS and Linux](https://aka.ms/getps6-linux)
 5. Download and install the Bot Framework (CLI) tool dependencies. It's important to do this even if you have earlier versions as the Virtual Assistant makes use of the latest capabilities: 
    ```
-   npm install -g botdispatch ludown luis-apis qnamaker luisgen
+   npm install -g botdispatch ludown luis-apis qnamaker luisgen botskills
    ```
 6. Download and install the [Azure Command Line Tools (CLI)](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?view=azure-cli-latest).
 7. Download and install the [Bot Framework Emulator](https://aka.ms/botframework-emulator).

--- a/docs/tutorials/typescript/skill.md
+++ b/docs/tutorials/typescript/skill.md
@@ -29,7 +29,7 @@
    * [Download PowerShell Core on macOS and Linux](https://aka.ms/getps6-linux)
 3. Install  Bot Framework (CLI) tool dependencies. It's important to do this even if you have earlier versions as the Skill template makes use of the latest capabilities: 
    ```
-   npm install -g botdispatch, ludown, luis-apis, qnamaker, luisgen
+   npm install -g botdispatch ludown luis-apis luisgen
    ```
 4. Install [Yeoman](http://yeoman.io)
    ```

--- a/docs/tutorials/typescript/virtualassistant.md
+++ b/docs/tutorials/typescript/virtualassistant.md
@@ -3,7 +3,7 @@
 **APPLIES TO:** ✅ SDK v4
 
 ## In this tutorial
-- [Quickstart: Creating your first Virtual Assistant (TypeScript)](#quickstart-creating-your-first-virtual-assistant-typescript)
+- [Quickstart: Create your first Virtual Assistant (TypeScript)](#quickstart-create-your-first-virtual-assistant-typescript)
   - [In this tutorial](#in-this-tutorial)
   - [Intro](#intro)
     - [Purpose](#purpose)
@@ -51,7 +51,7 @@ A Virtual Assistant app (in TypeScript) that greets a new user.
    * [Download PowerShell Core on macOS and Linux](https://aka.ms/getps6-linux)
 3. Download and install Bot Framework (CLI) tool dependencies. It's important to do this even if you have earlier versions as the Virtual Assistant makes use of the latest capabilities: 
    ```
-   npm install -g botdispatch, ludown, luis-apis, qnamaker, luisgen
+   npm install -g botdispatch ludown luis-apis qnamaker luisgen
    ```
 4. Install [Yeoman](http://yeoman.io)
    ```

